### DISCUSSION TITLE
global_args understanding

### DISCUSSION THREAD

**Discussion by k0stett** [roflcoopter/viseron#541] [Q&A] [answered]
I have a list of ffmpeg params to make ffmpeg not to crash while read stream from one shitty camera but I do not catch the magic how to write it down to viseron configuration file:
`-err_detect ignore_err -y -fflags +genpts -thread_queue_size 512 -use_wallclock_as_timestamps 1 `

Could somebody explain me that? I think it's gonna to global_args, but I cannot understand syntax for that.

**Comment by roflcoopter** [comment]
What does your full working command look like? 

Depending on where you put them you can use `global_args` or `input_args`

Config should look like this:
```yaml
input_args:
  - -err_detect 
  - ignore_err 
  - -y 
  - -fflags 
  - +genpts 
  - -thread_queue_size 
  - 512 
  - -use_wallclock_as_timestamps 
  - 1
```

Indentation needs to be fixed of course, wrote this out on my phone but I think I got it right

**Reply by k0stett** [reply]
> What does your full working command look like?

`ffmpeg -err_detect ignore_err -y -c:v h264 -fflags +genpts -thread_queue_size 512 -use_wallclock_as_timestamps 1 -timeout 5000000 -rtsp_transport udp -i rtsp://admin:QWerty123@10.0.0.125:554/onvif1 -dn -sn -map 0:0 -codec:v h264 -preset:v medium -b:v 16384k -maxrate 16384k -bufsize 16384k -r 15 -pix_fmt yuv420p -g 30 -map 0:1 -filter:a aresample=osr=44100:ochl=mono -codec:a aac -b:a 32k -shortest output.mp4`

May contains errors but it was a kind of pain to get video from it

**Reply by roflcoopter** [reply]
And what does the Viseron generated command look like if you use the `input_args` config that i wrote above?

You probably need to make some more changes to the config but its easier to take it one step at a time

**Reply by k0stett** [reply, marked as answer]
So I solved my problem in radical way: I've flashed camera with OpenIPC. Thanks for answers.
